src/geometry/geom2/canonicalize.js

```diff
@@ -0,0 +1,24 @@
+const vec2 = require('../../math/vec2')
+
+/**
+ * Produces a canonicalized geometry by canonicalizing the sides.
+ * Must be called before exposing any side data.
+ * @param {geom2} geometry - the geometry to canonicalize
+ * @returns {geom2} the given geometry with transformed points
+ * @example
+ * let newgeometry = canonicalize(geometry)
+ */
+const canonicalize = (geometry) => {
+  if (geometry.isCanonicalized) return geometry
+
+  // apply transforms to each side, and canonicalize
+  geometry.sides = geometry.baseSides.map((side) => {
```

Comment:
why basesides ?
storing the same (or almost same) data twice in immutable data structures does not make much sense
do not forget that we have in jscad :
- obj2 = operation(obj1)
- so we do not need to keep seperate structures (very costly copies) for sides that have been transformed (transformation here being canonicalized)

Reply:
gone. please review again.